### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by matnich89**
## Summary by CodeRabbit

- **New Features**
  - Introduced real-time train cancellation tracking and management service.
  - Added Redis integration for queue management.
  - Implemented worker functionality for processing train departure information.
  - Enhanced database schema to support detailed cancellation logging.

- **Bug Fixes**
  - Improved error handling during server shutdown.

- **Documentation**
  - Updated README with setup and usage instructions.
  - Included LICENSE.txt for software usage rights and disclaimers.

- **Chores**
  - Added handling for loading configurations from environment variables.

- **Refactor**
  - Enhanced signal handling for graceful server shutdown.
  - Renamed and updated database methods for cancellations.

### CODE REVIEW

File: db/db_test.go [hunk 1]
```diff
@@ -85,37 +85,45 @@ func TestCancellationDb(t *testing.T) {
...
-		assert.Equal(t, value, retrievedValue)
+		assert.Equal(t, 3, count)
 
-		nonExistentDate := "2000-01-01"
-		retrievedValue, err = testDb.GetValue(nonExistentDate)
+		differentDate := date.AddDate(0, 0, -1)
+		count, err = testDb.GetCancellationCountForDate(differentDate)
 		assert.NoError(t, err)
-		assert.Equal(t, 0, retrievedValue)
+		assert.Equal(t, 0, count)
```
[the remark this answers is not in the document]
Comment (matnich89) [reply db/db_test.go:104]: yes please draft them

**Comment by matnich89** [comment]
@coderabbitai can you generate a read me for this project?

File: model/model.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package model
+
+import "time"
+
+type DepartingTrainId struct {
+	ID string `json:"id"`
+}
+
+type Cancellation struct {
+	ID                 string
```
Comment (LesleySommerville-Ki) [inline model/model.go:10]: each cancellation has a unique key?

**Review by LesleySommerville-Ki — APPROVED** [review]
don't you ever get sick of `if err != nil {` 😂

File: model/model.go (hunk 2 above)
Comment (matnich89) [reply model/model.go:10]: Yeah , though its slightly useless right now, but the trainID is recycled every day ( this comes from national rail) so I couldn't use that, I have debating if a relational DB is actually right for this to be honest.

**Pull request merged**